### PR TITLE
✨ [Feature #498]: Add Error Boundaries

### LINKED ISSUE #498: ✨ [Feature request]: Add Error Boundaries
As the application grows, there might be chances that some components are error-prone and result in a blank page issue. so to overcome this issue we should add Error Boundaries that will render the fallback UI when there is an error and the user will be able to go back or try again.

> Error boundaries are React components that catch JavaScript errors anywhere in their child component tree, log those errors, and display a fallback UI instead of the component tree that crashed. [Read more](https://reactjs.org/docs/error-boundaries.html)

cc : @atapas @koustov

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

I worked on adding an error boundary to the application, I have used [https://www.npmjs.com/package/react-error-boundary](https://www.npmjs.com/package/react-error-boundary) package to achieve the same.
It's a quite popular package and has `2,444,018+` weekly downloads.

If interested you can read [Kent C. Dodds](https://kentcdodds.com/) article.
https://kentcdodds.com/blog/use-react-error-boundary-to-handle-errors-in-react

Fixes #498 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 
image

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Sachin-chaurasiya** [comment]
@atapas @koustov , PR is ready for review.

**Comment by Sachin-chaurasiya** [comment]
@atapas , Awaiting for your approval.

File: src/ErrorBoundary/ErrorFallback.jsx
Comment (koustov) [inline src/ErrorBoundary/ErrorFallback.jsx:27]: change the classname to error-boundary-image

File: src/index.js
Comment (koustov) [inline src/index.js:38]: I would prefer to have the navbar intact while showing errorboundary component. There are two options 
- We can re-create navbar with minimalistic (only the logo) elements and put inside error boundary component
- Create the error boundary at route level instead of root level

Do you agree?

**Review by koustov — CHANGES_REQUESTED** [review]
tiny comments

**Comment by Angryman18** [comment]
we already have error-boundary in react official docs but as class based component (`componentDidCatch` method doesnt have alternative in functional component). we can think of using it does the same thing instead of the external npm package. looking forward for other thoughts. https://reactjs.org/docs/error-boundaries.html

**Comment by Sachin-chaurasiya** [comment]
> we already have error-boundary in react official docs but as class based component (`componentDidCatch` method doesnt have alternative in functional component). we can think of using it does the same thing instead of the external npm package. looking forward for other thoughts. https://reactjs.org/docs/error-boundaries.html

Hey @Angryman18 , Thanks. the problem with this approach is that we have all components as functional components and
React boundary is only supported for in-class components and we don't want to go with the class component approach.

The package we are using it's quite popular and suggested by many react experts.
If interested you can read [Kent C. Dodds](https://kentcdodds.com/) article.
https://kentcdodds.com/blog/use-react-error-boundary-to-handle-errors-in-react

File: src/index.js
Comment (Sachin-chaurasiya) [reply src/index.js:38]: @koustov , I think we should not show the navbar when the page goes blank because it's irrelevant.
Error boundary should be at the root level so that we can catch all the errors.

**Comment by Angryman18** [comment]
> > we already have error-boundary in react official docs but as class based component (`componentDidCatch` method doesnt have alternative in functional component). we can think of using it does the same thing instead of the external npm package. looking forward for other thoughts. https://reactjs.org/docs/error-boundaries.html
> 
> Hey @Angryman18 , Thanks. the problem with this approach is that we have all components as functional components and React boundary is only supported for in-class components and we don't want to go with the class component approach.
> 
> The package we are using it's quite popular and suggested by many react experts. If interested you can read [Kent C. Dodds](https://kentcdodds.com/) article. https://kentcdodds.com/blog/use-react-error-boundary-to-handle-errors-in-react

my point of view was not using a 53KB cost package where we could able to do it by adding few lines of code. other thant that  that everything is fine. we can do it this way.

**Comment by Sachin-chaurasiya** [comment]
> > > we already have error-boundary in react official docs but as class based component (`componentDidCatch` method doesnt have alternative in functional component). we can think of using it does the same thing instead of the external npm package. looking forward for other thoughts. https://reactjs.org/docs/error-boundaries.html
> > 
> > 
> > Hey @Angryman18 , Thanks. the problem with this approach is that we have all components as functional components and React boundary is only supported for in-class components and we don't want to go with the class component approach.
> > The package we are using it's quite popular and suggested by many react experts. If interested you can read [Kent C. Dodds](https://kentcdodds.com/) article. https://kentcdodds.com/blog/use-react-error-boundary-to-handle-errors-in-react
> 
> thats fine but my point of view was not using a 56KB cost package where we could able to do it by adding few lines of code. other thant that that everything is fine. we can do it this way.

@Angryman18 , I got your point. but here we should look for maintainable code and not the package cost. It is used by millions of people to keep the code clear and maintainable.

**Comment by Sachin-chaurasiya** [comment]
@koustov @atapas, Awaiting for your approval.

**Review by atapas — APPROVED** [review]

**Comment by Sachin-chaurasiya** [comment]
@koustov , Waiting for your approval.

**Review by koustov — APPROVED** [review]
lgtm

**Pull request merged**
